client/ayon_core/hosts/houdini/api/hda_utils.py

```diff
@@ -0,0 +1,402 @@
+        folder_id=folder_entity["id"],
+        fields=id_only)
+    if not product_entity:
+        return []
+
+    # TODO: Support hero versions
+    versions = get_versions(project_name=project_name,
+                            product_ids=[product_entity["id"]],
+                            fields=["version"],
+                            hero=False)
+    version_names = [version["version"] for version in versions]
+    version_names.reverse()
+    return version_names
+
+
+def update_info(node, context):
+    """Update project, folder, product, version, representation name parms.
+
+     Arguments:
+         node (hou.Node): Node to update
+         context (dict): Context of representation
+
+     """
+    # TODO: Avoid 'duplicate' taking over the expression if originally
+    #       it was $OS and by duplicating, e.g. the `folder` does not exist
+    #       anymore since it is now `hero1` instead of `hero`
+    # TODO: Support hero versions
+    version = str(context["version"]["version"])
+
+    # We only set the values if the value does not match the currently
+    # evaluated result of the other parms, so that if the the project name
+    # value was dynamically set by the user with an expression or alike
+    # then if it still matches the value of the current representation id
+    # we preserve it
+    parms = {
+        "project_name": context["project"]["name"],
+        "folder_path": context["folder"]["path"],
+        "product_name": context["product"]["name"],
+        "version": version,
+        "representation_name": context["representation"]["name"],
+    }
+    parms = {key: value for key, value in parms.items()
+             if node.evalParm(key) != value}
+    parms["load_message"] = ""  # clear any warnings/errors
+
+    # Note that these never trigger any parm callbacks since we do not
+    # trigger the `parm.pressButton` and programmatically setting values
+    # in Houdini does not trigger callbacks automatically
+    node.setParms(parms)
+
+
+def _get_thumbnail(project_name, version_id, thumbnail_dir):
+    folder = hou.text.expandString(thumbnail_dir)
+    path = os.path.join(folder, "{}_thumbnail.jpg".format(version_id))
+    expanded_path = hou.text.expandString(path)
+    if os.path.isfile(expanded_path):
+        return path
+
+    # Try and create a thumbnail cache file
+    data = ayon_api.get_thumbnail(project_name,
+                                  entity_type="version",
+                                  entity_id=version_id)
+    if data:
+        thumbnail_dir_expanded = hou.text.expandString(thumbnail_dir)
+        os.makedirs(thumbnail_dir_expanded, exist_ok=True)
+        with open(expanded_path, "wb") as f:
+            f.write(data.content)
+        return path
+
+
+def set_representation(node, repre_id):
+    file_parm = node.parm("file")
+    if repre_id:
+        project_name = node.evalParm("project_name") or \
+                       get_current_project_name()
+        try:
```

Comment:
I believe this might be an option to take out the try-except. The reason why I think that might be important is that the try-except Exception might shadow other errors. This solution, together with what I placed into the lib.py, does survive my testing and removes the try-except.

you probably want to check the indent. my copy past doesn't carry it over
```suggestion
repre_entity = None
if lib.is_valid_id(repre_id):
        repre_entity = get_representation_by_id(project_name, repre_id)

```

Reply:
Thanks - I'll continue the conversation here: https://github.com/ynput/ayon-core/pull/294#discussion_r1547373265

Reply:
Implemented with https://github.com/ynput/ayon-core/pull/294/commits/bef67347ecbc990813a3b933c7c5342815bfabb7